diff --git a/yoga/CMakeLists.txt b/yoga/CMakeLists.txt
index b6eca1ac..42ab16e3 100644
--- a/yoga/CMakeLists.txt
+++ b/yoga/CMakeLists.txt
@@ -22,7 +22,7 @@ file(GLOB SOURCES CONFIGURE_DEPENDS
     ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/**/*.cpp)

-add_library(yogacore STATIC ${SOURCES})
+add_library(yogacore ${SOURCES})

 # Yoga conditionally uses <android/log> when building for Android
 if (ANDROID)
